File: src/easy/226-invert-binary-tree/invert_binary_tree.py
from typing import Optional

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root is None:
            return None
        
        def recurse(node) -> None:
            if not node:
                return
            
            left = node.left
            right = node.right

            node.left = right
            node.right = left

            if node.right:
                recurse(node.right)

            if node.left:
                recurse(node.left)
        
        recurse(root)
        return root

File: src/easy/226-invert-binary-tree/invertBinaryTree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
  if (!root) {
    return null;
  }

  var recurse = function(node) {
    if (!node) {
      return;
    }

    let left = node.left;
    let right = node.right;

    node.left = right;
    node.right = left;

    if (node.right) {
      recurse(node.right);
    }

    if (node.left) {
      recurse(node.left);
    }
  }

  recurse(root);

  return root;
};